### ISSUE TITLE
REST call to get all IP adresses which are known by one ifid

### ISSUE THREAD

**Issue by TimotheusBachinger** [ntop/ntopng#4081]
Hi ntop Team,

We need to map a list of checkmk's internal IP adresses to the known IP adresses of ntop ng.
This could be similar to: lua/hosts_stats.lua

We formerly discussed in our google docs with: "implement endpoint get/host/interfaces.lua which returns the list"

Regards
Timi

**Comment by simonemainardi** [comment]
what about `scripts/lua/rest/v1/get/interface/address.lua`? It should do what you are looking at.

**Comment by simonemainardi** [comment]
See also the documentation at: https://www.ntop.org/guides/ntopng/api/rest/examples_v1.html#get-interface-ip-addresses

**Comment by TimotheusBachinger** [comment]
Hi Simone, 

as far as we understood, `get/interface/address.lua` returns the IP adress to an ifid. E.g. in our system, where we only have one ifid (which is an zmq interface), we get no IP at all from this call. However in the ntopng web ui we see of course a lot of different IP adresses, which are monitored.
This is what we would like to receive via a REST call.

Regards
Timi

**Comment by simonemainardi** [comment]
> We formerly discussed in our google docs with: "implement endpoint get/host/interfaces.lua which returns the list"

This is implemented already: https://www.ntop.org/guides/ntopng/api/rest/examples_v1.html#get-host-interfaces

> in our system, where we only have one ifid (which is an zmq interface), we get no IP at all from this call

This is normal. `get/interface/address.lua` returns the physical network interface address as read from the operating system and there's no such address for ZMQ.

> However in the ntopng web ui we see of course a lot of different IP adresses, which are monitored.
This is what we would like to receive via a REST call.

To retrieve the list of active hosts of a particular interface the endpoint is: https://www.ntop.org/guides/ntopng/api/rest/examples_v1.html#get-active-hosts

----

So basically is seems there's everything you need. Alternatively, please explain with an example.

**Comment by simonemainardi** [comment]
@TimotheusBachinger please provide feedback

**Comment by TimotheusBachinger** [comment]
Hi Simone,

sorry for not replying earlier. https://www.ntop.org/guides/ntopng/api/rest/examples_v1.html#get-active-hosts is what were looking for. 
Thanks a lot!

Regards 
Timi

**Comment by simonemainardi** [comment]
good, thanks for reporting
